# Preserves: export RK_MISC=wipe_all-misc.img
diff --git a/project/cfg/BoardConfig_IPC/BoardConfig-SPI_NAND-NONE-RV1106_IPC38_DEMO_V10-IPC-XWR60440.mk b/project/cfg/BoardConfig_IPC/BoardConfig-SPI_NAND-NONE-RV1106_IPC38_DEMO_V10-IPC-XWR60440.mk
index 118d4cd..7a770db 100644
--- a/project/cfg/BoardConfig_IPC/BoardConfig-SPI_NAND-NONE-RV1106_IPC38_DEMO_V10-IPC-XWR60440.mk
+++ b/project/cfg/BoardConfig_IPC/BoardConfig-SPI_NAND-NONE-RV1106_IPC38_DEMO_V10-IPC-XWR60440.mk
@@ -52 +52 @@
-export RK_PARTITION_CMD_IN_ENV="256K(env),1M@256K(idblock),1M(uboot),4M(boot),32M(rootfs),48M(oem),-(userdata)"
+export RK_PARTITION_CMD_IN_ENV="256K(env),1M@256K(idblock),1M(uboot),4M(boot_a),4M(boot_b),10M(rootfs_a),10M(rootfs_b),32M(oem_a),32M(oem_b),1792K(reserved),256K(misc),-(userdata)"
@@ -64 +64 @@
-export RK_PARTITION_FS_TYPE_CFG=rootfs@IGNORE@squashfs,oem@/oem@squashfs,userdata@/userdata@ubifs
+export RK_PARTITION_FS_TYPE_CFG=rootfs_a@IGNORE@squashfs,oem_a@/oem@squashfs,userdata@/userdata@ubifs
