diff --git a/sysdrv/source/uboot/u-boot/include/linux/mtd/spinand.h b/sysdrv/source/uboot/u-boot/include/linux/mtd/spinand.h
index b00a1e8..4470d1d 100755
--- a/sysdrv/source/uboot/u-boot/include/linux/mtd/spinand.h
+++ b/sysdrv/source/uboot/u-boot/include/linux/mtd/spinand.h
@@ -158,0 +159 @@
+#define BL_LOWER_3_4_LOCKED	0x2a
@@ -319,0 +321 @@ struct spinand_ecc_info {
+#define SPINAND_HAS_BLOCK_PROTECTION BIT(2)
@@ -512,0 +515,2 @@ int spinand_select_target(struct spinand_device *spinand, unsigned int target);
+int spinand_get_block_lock(struct spinand_device *spinand, u8 *lock);
+int spinand_set_block_lock(struct spinand_device *spinand, u8 lock);
diff --git a/sysdrv/source/uboot/u-boot/drivers/mtd/nand/spi/core.c b/sysdrv/source/uboot/u-boot/drivers/mtd/nand/spi/core.c
index 46b028d..cd0570c 100755
--- a/sysdrv/source/uboot/u-boot/drivers/mtd/nand/spi/core.c
+++ b/sysdrv/source/uboot/u-boot/drivers/mtd/nand/spi/core.c
@@ -495,0 +496,21 @@ static int spinand_lock_block(struct spinand_device *spinand, u8 lock)
+int spinand_get_block_lock(struct spinand_device *spinand, u8 *lock)
+{
+	return spinand_read_reg_op(spinand, REG_BLOCK_LOCK, lock);
+}
+
+int spinand_set_block_lock(struct spinand_device *spinand, u8 lock)
+{
+	u8 readback;
+	int ret;
+
+	ret = spinand_write_reg_op(spinand, REG_BLOCK_LOCK, lock);
+	if (ret)
+		return ret;
+
+	ret = spinand_get_block_lock(spinand, &readback);
+	if (ret)
+		return ret;
+
+	return readback == lock ? 0 : -EIO;
+}
+
@@ -1275 +1296,5 @@ static int spinand_init(struct spinand_device *spinand)
-		ret = spinand_lock_block(spinand, BL_ALL_UNLOCKED);
+		if (spinand->flags & SPINAND_HAS_BLOCK_PROTECTION) {
+			ret = spinand_set_block_lock(spinand, BL_LOWER_3_4_LOCKED);
+		} else {
+			ret = spinand_set_block_lock(spinand, BL_ALL_UNLOCKED);
+		}
diff --git a/sysdrv/source/uboot/u-boot/drivers/mtd/nand/spi/chucun.c b/sysdrv/source/uboot/u-boot/drivers/mtd/nand/spi/chucun.c
index c4f0d34..114613a 100755
--- a/sysdrv/source/uboot/u-boot/drivers/mtd/nand/spi/chucun.c
+++ b/sysdrv/source/uboot/u-boot/drivers/mtd/nand/spi/chucun.c
@@ -89 +89 @@ static const struct spinand_info chucun_spinand_table[] = {
-		     SPINAND_HAS_QE_BIT,
+		     SPINAND_HAS_QE_BIT | SPINAND_HAS_BLOCK_PROTECTION,
@@ -98 +98 @@ static const struct spinand_info chucun_spinand_table[] = {
-		     SPINAND_HAS_QE_BIT,
+		     SPINAND_HAS_QE_BIT | SPINAND_HAS_BLOCK_PROTECTION,
